src/utils/BlockUtils.ts

```diff
@@ -0,0 +1,53 @@
+import { BlockStruct } from "@typechain-types/contracts/V1/DataTypes";
+import { AddressLike } from "ethers";
+
+/**
+ * Extract numeric fields from a block and convert them to regular number types
+ */
+export const coordinatesOf = (block: BlockStruct) => ({
+    forkCnt: Number(block.transaction.header.forkCnt),
+    height: Number(block.transaction.header.transactionCnt)
+});
+
+/**
+ * Get the block height (transaction count) from a block
+ */
+export const heightOf = (block: BlockStruct): number =>
+    Number(block.transaction.header.transactionCnt);
+
+/**
+ * Get the fork number from a block
+ */
+export const forkOf = (block: BlockStruct): number =>
+    Number(block.transaction.header.forkCnt);
+
+/**
+ * Get the timestamp from a block
+ */
+export const timestampOf = (block: BlockStruct): number =>
+    Number(block.transaction.header.timestamp);
+
+export const participantOf = (block: BlockStruct): AddressLike =>
+    block.transaction.header.participant;
```

Comment:
Normalize this too and if the "Like" type is needed just access the block directly